### PR TITLE
LPS-169305 Do not calculate category usage count if the field is not requested

### PR DESCRIPTION
**Pull request by LuismiBarcos**
Related ticket -> [LPS-169305](https://issues.liferay.com/browse/LPS-169305)
____

This PR made a quick check to see if the property `taxonomyCategoryUsageCount` is in the query param `restrictFields`. If it is there, there is no need to calculate the property, improving the performance.

### CODE REVIEW

**Comment by LuismiBarcos** [comment]
ci:test:sf

**Comment by LuismiBarcos** [comment]
ci:test:relevant

**Comment by LuismiBarcos** [comment]
I would like to do it with streams but as this is a priority, I did it like it is.

File: modules/apps/headless/headless-admin-taxonomy/headless-admin-taxonomy-impl/src/main/java/com/liferay/headless/admin/taxonomy/internal/dto/v1_0/converter/TaxonomyCategoryDTOConverter.java [hunk 1]
```diff
@@ -169,6 +164,50 @@ private TaxonomyCategory _toTaxonomyCategory(
...
+						if (uriInfoOptional.isPresent()) {
+							UriInfo uriInfo = uriInfoOptional.get();
+
+							MultivaluedMap<String, String> queryParameters =
+								uriInfo.getQueryParameters();
+
+							String restrictFields = queryParameters.getFirst(
+								"restrictFields");
+
+							String[] restrictFieldsArray = restrictFields.split(
```
Comment (ccorreagg) [inline modules/apps/headless/headless-admin-taxonomy/headless-admin-taxonomy-impl/src/main/java/com/liferay/headless/admin/taxonomy/internal/dto/v1_0/converter/TaxonomyCategoryDTOConverter.java:185]: Maybe better to use this method from the `StringUtil` class:

`public static List<String> split(String s)`

`restictFieldsList = StringUtil.split(restrictFields);`

Comment (ccorreagg) [reply modules/apps/headless/headless-admin-taxonomy/headless-admin-taxonomy-impl/src/main/java/com/liferay/headless/admin/taxonomy/internal/dto/v1_0/converter/TaxonomyCategoryDTOConverter.java:185]: In fact, I think we could do something more magical :magic_wand: to use less variables: 

`						
						Optional<UriInfo> uriInfoOptional =
							dtoConverterContext.getUriInfoOptional();

						if (uriInfoOptional.isPresent()) {
							UriInfo uriInfo = uriInfoOptional.get();

							MultivaluedMap<String, String> queryParameters =
								uriInfo.getQueryParameters();

							String restrictFields = queryParameters.getFirst(
								"restrictFields");

							if (ListUtil.exists(
									StringUtil.split(restrictFields),
									"taxonomyCategoryUsageCount"::equals)) {

								return null;
							}
						}
`

Maybe too magical :thinking:

Comment (LuismiBarcos) [reply modules/apps/headless/headless-admin-taxonomy/headless-admin-taxonomy-impl/src/main/java/com/liferay/headless/admin/taxonomy/internal/dto/v1_0/converter/TaxonomyCategoryDTOConverter.java]: True! Done! Thanks Carlos 🙂

**Comment by LuismiBarcos** [comment]
ci:test:sf

File: modules/apps/headless/headless-admin-taxonomy/headless-admin-taxonomy-impl/src/main/java/com/liferay/headless/admin/taxonomy/internal/dto/v1_0/converter/TaxonomyCategoryDTOConverter.java (hunk 1 above)
Comment (LuismiBarcos) [reply modules/apps/headless/headless-admin-taxonomy/headless-admin-taxonomy-impl/src/main/java/com/liferay/headless/admin/taxonomy/internal/dto/v1_0/converter/TaxonomyCategoryDTOConverter.java]: Well, my first implementation was something like
```
List<String> restrictFields = uriInfoOptional
							.map(UriInfo::getQueryParameters)
							.map(queryParameters -> queryParameters.getFirst(
								"restrictFields"))
							.map(StringUtil::split)
							.map(Arrays::asList)
							.orElse(Collections.emptyList());
						
						if(restrictFields.contains("taxonomyCategoryUsageCount")) {
```

But the I remembered that Brian's does not like streams, so... 🤷‍♂️

**Comment by brianchandotcom** [comment]
@LuismiBarcos please verify my simplification in https://github.com/brianchandotcom/liferay-portal/commit/e2e2b6969376129cefcfc4fd3ef701bf47d07770

I'm relying on

```
	/**
	 * Returns <code>true</code> if the string contains the text as one or more
	 * consecutive comma delimited list entries.
	 *
	 * <p>
	 * Examples:
	 * </p>
	 *
	 * <p>
	 * <pre>
	 * <code>
	 * contains("one,two,three", "two") returns true
	 * contains("one,two,three", "thr") returns false
	 * contains("one,two,three", "one,two") returns true
	 * </code>
	 * </pre></p>
	 *
	 * @param  s the string in which to search
	 * @param  text the text to search for in the string
	 * @return <code>true</code> if the string contains the text as one or more
	 *         consecutive comma delimited list entries; <code>false</code>
	 *         otherwise
	 */
	public static boolean contains(String s, String text) {
		return contains(s, text, StringPool.COMMA);
	}
```

**Comment by LuismiBarcos** [comment]
Closing this PR as was merged by Brian
https://github.com/brianchandotcom/liferay-portal/commit/998a3e89051632fae77efc2a26f4f9ee8703fd74
https://github.com/brianchandotcom/liferay-portal/commit/e2e2b6969376129cefcfc4fd3ef701bf47d07770

**Pull request closed without merging**